### ISSUE TITLE
similar_names not triggered by example

### ISSUE THREAD

**Issue by ASBalon** [rust-lang/rust-clippy#16110]
### Summary

https://rust-lang.github.io/rust-clippy/master/index.html?groups=pedantic&search=Similar_names

The example listed on the clippy index no longer triggers similar_name lint.

The example listed on the index should be included it the unit testing for this lint.

### Lint Name

similar_names

### Reproducer

I tried this code:

```rust

let checked_exp = something;
let checked_expr = something_else;

```

I expected to see this happen:
Lint warning

Instead, this happened:
Nothing 

### Version

```text

```
